Sources/StreamChatUI/ChatChannel/ChatChannelVC.swift

```diff
@@ -249,13 +267,22 @@ open class ChatChannelVC: _ViewController,
         _ channelController: ChatChannelController,
         didUpdateMessages changes: [ListChange<ChatMessage>]
     ) {
-        if isLastMessageFullyVisible {
+        if shouldMarkChannelRead {
             channelController.markRead()
         }
 
+        // In order to not show an empty list when jumping to a message, ignore the remove updates.
+        if !isFirstPageLoaded && changes.filter(\.isRemove).count == messages.count {
+            return
+        }
+
         messageListVC.setPreviousMessagesSnapshot(messages)
         messageListVC.setNewMessagesSnapshot(Array(channelController.messages))
-        messageListVC.updateMessages(with: changes)
+        messageListVC.updateMessages(with: changes) { [weak self] in
+            if self?.shouldMarkChannelRead == true {
+                self?.channelController.markRead()
```

Comment:
Is this something that we need to guard and make sure that markRead will run on the MainThread for UI changes? (sorry cannot find the path to this method to see what and how it updates the UI)

Reply:
This is always run on the main thread